presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -115,6 +118,20 @@ public Session getSession()
         return session;
     }
 
+    public long getPeakMemoryInBytes()
+    {
+        return peakMemory.get();
+    }
+
+    public void updateMemoryUsage(long deltaMemoryInBytes)
+    {
+        currentMemory.addAndGet(deltaMemoryInBytes);
+        long currentMemoryValue = currentMemory.get();
```

Comment:
There's still a race here. Another thread might decrement `currentMemory` between the call to `addAndGet` and `get`